core/src/main/scala/kafka/controller/TopicDeletionManager.scala

```diff
@@ -270,9 +271,7 @@ class TopicDeletionManager(config: KafkaConfig,
     }
 
     client.sendMetadataUpdate(partitions)
-    topics.foreach { topic =>
-      onPartitionDeletion(controllerContext.partitionsForTopic(topic))
-    }
+    onPartitionDeletion(partitions)
```

Comment:
In this function, we start with a list of topics and convert it to a flattened list of partitions. Ultimately in `startReplicaDeletion`, we end up regrouping by topic. I wonder if there is an opportunity to save a little work by avoiding the intermediate conversions.

By the way, because we are doing more batching now, this may be a good time to reduce or remove the `info` log line in `onPartitionDeletion` which can already be a bit noisy.

Reply:
Good point. I have made an attempt which keeps a Map all the way through. Let me know what do you think.

Reply:
I have removed the log as well.